main.tf

```
@@ -1787,6 +1787,113 @@ module "secrets_store_csi_driver" {
+  max_history                = try(var.aws_privateca_issuer.max_history, null)
+  atomic                     = try(var.aws_privateca_issuer.atomic, null)
+  skip_crds                  = try(var.aws_privateca_issuer.skip_crds, null)
+  render_subchart_notes      = try(var.aws_privateca_issuer.render_subchart_notes, null)
+  disable_openapi_validation = try(var.aws_privateca_issuer.disable_openapi_validation, null)
+  wait                       = try(var.aws_privateca_issuer.wait, null)
+  wait_for_jobs              = try(var.aws_privateca_issuer.wait_for_jobs, null)
+  dependency_update          = try(var.aws_privateca_issuer.dependency_update, null)
+  replace                    = try(var.aws_privateca_issuer.replace, null)
+  lint                       = try(var.aws_privateca_issuer.lint, null)
+
+  postrender = try(var.aws_privateca_issuer.postrender, [])
+  set = concat([
+    {
+      name  = "serviceAccount.name"
+      value = local.aws_privateca_issuer_service_account
+    }],
+    try(var.aws_privateca_issuer.set, [])
+  )
+  set_sensitive = try(var.aws_privateca_issuer.set_sensitive, [])
+
+  # IAM role for service account (IRSA)
+
+  set_irsa_names                = ["serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"]
+  create_role                   = try(var.aws_privateca_issuer.create_role, true)
+  role_name                     = try(var.aws_privateca_issuer.role_name, "aws-privateca-issuer")
+  role_name_use_prefix          = try(var.aws_privateca_issuer.role_name_use_prefix, true)
+  role_path                     = try(var.aws_privateca_issuer.role_path, "/")
+  role_permissions_boundary_arn = lookup(var.aws_privateca_issuer, "role_permissions_boundary_arn", null)
+  role_description              = try(var.aws_privateca_issuer.role_description, "IRSA for aws-privateca-issuer")
+  role_policies                 = lookup(var.aws_privateca_issuer, "role_policies", {})
+
+  source_policy_documents = compact(concat(
+    data.aws_iam_policy_document.aws_privateca_issuer[*].json,
+    lookup(var.aws_privateca_issuer, "source_policy_documents", [])
+  ))
+  override_policy_documents = lookup(var.aws_privateca_issuer, "override_policy_documents", [])
+  policy_statements         = lookup(var.aws_privateca_issuer, "policy_statements", [])
+  policy_name               = try(var.aws_privateca_issuer.policy_name, "aws-privateca-issuer")
+  policy_name_use_prefix    = try(var.aws_privateca_issuer.policy_name_use_prefix, true)
+  policy_path               = try(var.aws_privateca_issuer.policy_path, null)
+  policy_description        = try(var.aws_privateca_issuer.policy_description, "IAM Policy for AWS PCA Issuer")
+
+  oidc_providers = {
+    controller = {
+      provider_arn = var.oidc_provider_arn
+      # namespace is inherited from chart
+      service_account = local.aws_privateca_issuer_service_account
+    }
+  }
+
+  tags = var.tags
+
+}
+
+data "aws_iam_policy_document" "aws_privateca_issuer" {
+  statement {
```

add a count here so it doesn't show up in plans unless users have enabled it:

```suggestion
  count = var.enable_aws_privateca_issuer ? 1 : 0
```